compiler/rustc_passes/messages.ftl

```diff
@@ -804,6 +804,9 @@ passes_unused_duplicate =
 passes_unused_empty_lints_note =
     attribute `{$name}` with an empty list has no effect
 
+passes_unused_linker_warnings_note =
+    the `linker_warnings` lint can only be controlled at the root of a crate with `--crate-type bin`
```

Comment:
It is also allowed at the root of dylib, cdylib and proc-macro crates. Maybe say that it can only be controlled at the root of a crate that needs to be linked. Or just always allow it. If you define a crate as both rlib and dylib in Cargo.toml and compile for a target that doesn't support dylibs, cargo will drop the dylib crate type. This probably shouldn't cause a warning that `#[allow(linker_messages)]` is not allowed.

Reply:
i've updated the message. i don't want to remove this warning, i would rather get false positives about the attribute being ignored than have it silently discarded for lib crates that are never dylibs.